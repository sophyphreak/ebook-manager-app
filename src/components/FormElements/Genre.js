import React, { Component } from 'react';
import SelectInput from './SelectInput';

const Genre = ({
  isGenreDisabled,
  ...props
}) => (
  <SelectInput
    label="Genre:*"
    options={[
      'Please select',
      'Romance', 
      'Erotica',
      'Mystery/Thriller',
      'Science Fiction',
      'Young Adult',
      'Literary Fiction',
      'Children'
    ]}
    {...props}
  />
);

export default Genre; 