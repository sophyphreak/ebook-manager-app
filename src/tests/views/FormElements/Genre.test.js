import React from 'react';
import { shallow } from 'enzyme';
import Genre from '../../../views/FormElements/Genre';

let wrapper;

beforeEach(() => {
  wrapper = shallow(<Genre />);
});

test('should render Genre correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
