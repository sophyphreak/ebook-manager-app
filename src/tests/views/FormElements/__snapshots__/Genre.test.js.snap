// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Genre correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Genre />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Genre:*",
        <input
          onChange={undefined}
          type="text"
          value={undefined}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      "Genre:*",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": undefined,
          "type": "text",
          "value": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Genre:*",
          <input
            onChange={undefined}
            type="text"
            value={undefined}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Genre:*",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": undefined,
            "type": "text",
            "value": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
